contrib/starrocks-python-client/docs/test/test_coverage.md

```diff
@@ -8,82 +8,212 @@ This suite verifies that Python schema definitions are correctly compiled into S
+- **Autogenerate**: Detection of changes in all the properties listed above.
+
+### 6.2 Bitmap Index Lifecycle
+
+- `CREATE INDEX ... USING BITMAP` compilation and execution.
+- `DROP INDEX` compilation and execution.
+- Reflection of Bitmap indexes via `inspector.get_indexes()`.
+- `autogenerate` detection for the creation and deletion of Bitmap indexes.
+
+### 6.3 `AUTO_INCREMENT` Property
+
+- Full reflection of the `AUTO_INCREMENT` property on columns. Current tests are marked as `xfail`.
+- `autogenerate` comparison to produce an `ALTER` operation for adding/removing `AUTO_INCREMENT` (currently only a warning is logged).
+
+### 6.4 Complex Mixed-Operation Scenarios
+
+While individual operations are well-tested, the following complex scenarios need coverage:
+
+- Migration scripts that combine multiple, different types of operations in a single revision (e.g., adding a table, altering a column, dropping a view, and modifying MV properties all in one migration).
+- Cross-object dependencies in migrations (e.g., dropping a table that a view depends on).
+- Schema-qualified objects across multiple schemas in a single migration.
+
+### 6.5 Performance and Scale Testing
+
+- Large-scale migrations with many objects (e.g., 100+ tables).
+- Migration performance benchmarks.
+- Handling of very large table definitions with many columns and complex types.
+
+## 7. Test Organization Summary
+
+The test suite is organized in a pyramid structure, from unit tests to end-to-end system tests:
+
+```text
+                    ┌─────────────────────────┐
+                    │   System Tests (E2E)    │  ← Full Alembic lifecycle
+                    │   test/system/          │     with real database
+                    └─────────────────────────┘
+                  ┌───────────────────────────────┐
+                  │   Integration Tests           │  ← Database operations
+                  │   test/integration/           │     (reflection, autogenerate)
+                  └───────────────────────────────┘
+              ┌─────────────────────────────────────────┐
+              │   Unit Tests                            │  ← Individual components
+              │   test/unit/ & test/sql/                │     (comparison, rendering)
+              └─────────────────────────────────────────┘
+```
+
+**Key Testing Principles:**
+
+1. **Unit Tests** (`test/unit/`, `test/sql/`): Fast, isolated tests for individual functions and components.
+2. **Integration Tests** (`test/integration/`): Test interactions with a real StarRocks database (reflection, autogenerate comparison).
+3. **System Tests** (`test/system/`): End-to-end tests simulating the complete user workflow from schema definition through migration execution and rollback.
+
+The **system tests** are particularly important as they validate the entire Alembic lifecycle that users will experience in production, ensuring that:
```

Comment:
Probably should extend from the Alembic Test Suite provided by alembic - which integrates with the SQLAlchemy test suite, then all tests can be ran together?